src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -226,60 +231,80 @@ def read_cell_language(self, lines: list[str]) -> CellLanguage | None:
             return None
         return None
 
-    def new_cell(self, source: str) -> Cell:
-        return self._new_cell(source)
+    def new_cell(self, source: str, original_offset: int) -> Cell:
+        return self._new_cell(source, original_offset)
 
     def extract_cells(self, source: str) -> list[Cell] | None:
         lines = source.split('\n')
         header = f"{self.comment_prefix} {NOTEBOOK_HEADER}"
         if not lines[0].startswith(header):
             raise ValueError("Not a Databricks notebook source!")
 
-        def make_cell(cell_lines: list[str]):
+        def make_cell(cell_lines: list[str], start: int):
             # trim leading blank lines
-            while len(cell_lines) > 0 and len(cell_lines[0]) == 0:
-                cell_lines.pop(0)
+            new_cell_lines = cell_lines
```

Comment:
```suggestion
            new_cell_lines = cell_lines[:]
```
You wanted a copy, right?

Reply:
I assumed that leading and trailing blank lines were being deleted for a reason. The new_cell lines thing was a bug left over from refactor, caused by needing to fix a bug in the notebook parser that indeed corrupted the source:

```
# MAGIC %sql  SELECT * FROM 
```

Reply:
I iwll change some of the tests to change leading and trailing blanks lines form other than one